frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -36,14 +51,46 @@ const ClusterUpdateModal = withHandlePromise((props: ClusterUpdateModalProps) =>
       ? currentMinorVersionPatchUpdate?.version
       : availableSortedUpdates[0]?.version) || '',
   );
+  const [machineConfigPools, machineConfigPoolsLoaded] = useK8sWatchResource<
+    MachineConfigPoolKind[]
+  >({
+    isList: true,
+    kind: referenceForModel(MachineConfigPoolModel),
+  });
   const [error, setError] = React.useState(errorMessage);
   const [isOpen, setIsOpen] = React.useState(false);
-  const onToggle = () => setIsOpen(!isOpen);
-  const onSelect = (event, selection) => {
+  const [machineConfigPoolsToPause, setMachineConfigPoolsToPause] = React.useState<string[]>([]);
+  const [upgradeType, setUpgradeType] = React.useState<upgradeTypes>(upgradeTypes.Full);
+  React.useEffect(() => {
+    const initialMCPPausedValues = [];
+    machineConfigPools.map((mcp) => {
+      !isMCPMaster(mcp) && isMCPPaused(mcp)
+        ? initialMCPPausedValues.push(mcp.metadata.name)
+        : _.pull(mcp.metadata.name);
+    });
+    setMachineConfigPoolsToPause(initialMCPPausedValues);
+  }, [machineConfigPools]);
+  const onToggleVersion = () => setIsOpen(!isOpen);
+  const onSelectVersion = (event, selection) => {
     event.preventDefault();
     setDesiredVersion(selection);
     setIsOpen(!isOpen);
   };
+  const handleUpgradeTypeChange = (_checked: boolean, event: React.FormEvent<HTMLInputElement>) => {
+    const id = event?.currentTarget?.id;
+    setUpgradeType(id as typeof upgradeType);
```

Comment:
What if `id` is not available... you go from a value to undefined. Is this okay?

Reply:
In theory it shouldn't.  Are you raising the concern because of the (probably unnecessary) optional chaining on 80?  What is the correct approach?

Reply:
`Radio` is a PF component. `event` will not be `undefined` or `null` (or PF has a bug, not us). `currentTarget` is always going to be there because it's an event. `id` is set by your component, which I'll admit is not the best for React components (`id`s are intended to be unique in the whole HTML document and thus don't play nicely with React components that can render multiple times). So they are entirely not optionally chained because it'll all be there.

However, my recommendation is if you're unsure of where this data is coming from or how you want to deliver such a piece of information in a guaranteed fashion... use closures so you have full control.

Eg make your component:
```
<Radio onChange={(checked) => handleRadioOnChange(checked, SOME_CONSTANT)} />
```
That way you have just wrapped your callback in the value it needs without trying to pass a value through the DOM to your own callback.